### PR TITLE
BLD, MAINT: more build system cleanups

### PR DESCRIPTION
**Pull request by tylerjereddy**
* a collection of additional cleanups to the build system now that we are using only one (`meson`) and not depending on NumPy distutils/`setup.py` stuff

[skip cirrus]

### CODE REVIEW

**Review by ilayn — APPROVED** [review]

**Comment by rgommers** [comment]
This all looks good at first sight - guess we need greener CI before merging.

**Comment by starslabs** [comment]
您的邮件我已收到，我会加班加点的看完您珍贵的来信，尽快给您回复的！谢谢啦！陈鑫

**Comment by tylerjereddy** [comment]
This effectively has two core dev approvals and CI is passing after Ralf banned Cython `3.0.3`, so I think I'm allowed to self-merge.

**Pull request merged**
